jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -1076,6 +1076,115 @@ YUI.add('juju-env-api', function(Y) {
     },
 
     /**
+      Sends the provided resource info up to Juju without making it available
+      yet.
+
+      This is required to be called before deploying charms requiring
+      resources. The resource identifiers passed to the callback must then be
+      used when calling deploy. Like this:
+
+        const resources = [
+          // Retrieve resources from charmstore, for instance from
+          // https://api.jujucharms.com/charmstore/v5/$id/meta/resources
+        ];
+        model.addPendingResources({
+          applicationName: 'wordpress',
+          charmURL: 'wordpress-42',
+          channel: 'stable',
+          resources: resources
+        }, (err, ids) => {
+          if (err) {
+            // Handle error.
+            return;
+          }
+          model.deploy({
+            applicationName: 'wordpress',
+            charmURL: 'wordpress-42',
+            resources: ids
+          }, deployCallback);
+        });
+
+      @method addPendingResources
```

Comment:
I just recently learnt that with the new versions of jsdoc we don't have to manually specify the method names any longer.